compiler/src/CLI_errors.ml

```diff
@@ -0,0 +1,60 @@
+open Glob_options
+open Utils
+
+(* -------------------------------------------------------------------- *)
+
+type input_error =
+  | RedundantInputFile of string
+  | FileNotFound of string
+  | FileIsDirectory of string
+
+let pp_input_error ie =
+  match ie with
+  | RedundantInputFile s ->
+      Printf.sprintf
+        "Input file already set to %s, second input file %s is redundant"
+        !infile
+        s
+  | FileNotFound s -> Printf.sprintf "File %s not found" s
+  | FileIsDirectory s -> Printf.sprintf "File %s is a directory" s
```

Comment:
Please use `Format.asprintf`.

Reply:
Will do. What's the difference?

Reply:
Hmm. Now that you ask, I’m no longer sure. Sorry about that. Feel free to ignore the previous comment.

Reply:
I was told that `Format` format strings are more powerful, so to use them when I want more power. I have no opinion on this, and have been working on improvements are modularization to the assembly printing as well, so I'm interested in knowing if I should always use `Format` or sometimes one and sometimes the other.
Anyways, I simplified this PR and will push later.

Reply:
I will stick to `Format` everywhere.